devApps/AdalDesktopTestApp/MyHttpClientFactory.cs

```diff
@@ -0,0 +1,31 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved. 
+// Licensed under the MIT License.
+
+using Microsoft.IdentityModel.Clients.ActiveDirectory;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net.Http;
+using System.Net.Http.Headers;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace AdalDesktopTestApp
+{
+    class MyHttpClientFactory : IHttpClientFactory
+    {
+        private HttpClient _httpClient;
+
+        public HttpClient GetHttpClient()
+        {
+            _httpClient = new HttpClient(new HttpClientHandler() { UseDefaultCredentials = true })
```

Comment:
Suggestion: return the same httpclient for the lifetime of the factory? If you need something that creates a new object for testing purposes, you can add a factory method which creates a new one (CreateHttpClient)

Reply:
You can use Lazy<T> if you want